app/src/main/java/nodomain/freeyourgadget/gadgetbridge/GBApplication.java

```diff
@@ -247,6 +252,41 @@ public static boolean isRunningLollipopOrLater() {
         return VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
     }
 
+    public static boolean isRunningMarshmallowOrLater() {
+        return VERSION.SDK_INT >= Build.VERSION_CODES.M;
+    }
+
+    public static boolean isPriorityNumber(int prioritySenders, String number) {
+        Uri uri = Uri.withAppendedPath(PhoneLookup.CONTENT_FILTER_URI, Uri.encode(number));
+        String[] projection = new String[]{PhoneLookup._ID, PhoneLookup.STARRED};
+        Cursor cursor = context.getContentResolver().query(uri, projection, null, null, null);
+        boolean exists = false;
+        int starred = 0;
+        try {
+            if (cursor.moveToFirst()) {
+                exists = true;
+                starred = cursor.getInt(cursor.getColumnIndexOrThrow(PhoneLookup.STARRED));
+            }
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+        switch (prioritySenders) {
+            case Policy.PRIORITY_SENDERS_ANY:
```

Comment:
This case could be moved to the beginning of the method, so that the lookup is skipped altogether.